src/devices.jl

```diff
@@ -12,11 +12,24 @@ abstract type AbstractDevice end
 """
     CPUDevice()
 
-Use the CPU.
+Abstract device type for single-threaded and multi-threaded CPU runs.
 """
-struct CPUDevice <: AbstractDevice end
+abstract type CPUDevice <: AbstractDevice end
```

Comment:
```suggestion
abstract type AbstractCPUDevice <: AbstractDevice end
```
Should we use the `Abstract` prefix here?

Reply:
I think the name is perhaps better, and it has the benefit of us being able to deprecate `CPUDevice`

Reply:
I think this conversation was accidentally resolved

Reply:
Ah, the last commit missed the type @sriharshakandala

Reply:
Sorry, unexpected find/replace error. Its updated now!